Terminal.Gui/Core/ConsoleDriver.cs

```diff
@@ -90,7 +170,7 @@ public enum Color {
 	///   scenarios, they encode both the foreground and the background color and are used in the <see cref="ColorScheme"/>
 	///   class to define color schemes that can be used in your application.
 	/// </remarks>
-	public struct Attribute {
+	public class Attribute : IEquatable<Attribute> {
```

Comment:
Technically this is a breaking API change. This PR must be marked as "BREAKING CHANGE" and we need to determine whether this will actually break existing code or not.

Reply:
We have to distinguish between basic colors and extended colors. This is necessary to decide which escape code format to use later. The easiest way to do this is to change the color attribute struct to a class.